src/schnorr/ed25519.ts

```diff
@@ -115,6 +139,17 @@ export class PublicKeyWithChainCode {
 export class DerivationPath {
 	constructor(public readonly path: PathComponent[]) {}
 
+	/**
+	 * Creates a new DerivationPath from a canister identifier plus other path components
+	 * @param canisterId the id of the canister
+	 * @param path other path components
+	 */
+	static withCanisterPrefix(canisterId: Principal, path: PathComponent[]) {
```

Comment:
```suggestion
	static withCanisterPrefix(canisterId: Principal, path: PathComponent[]): DerivationPath {
```
Specifying the return type might be helpful.